src/components/page/Member/MemberDetail.tsx

```diff
@@ -1,29 +1,24 @@
 import { useRouter } from "next/router";
-import { useEffect } from "react";
 import { Group, Spoiler, Text } from "@mantine/core";
+import { useQuery } from "@tanstack/react-query";
 import { AppLoading } from "src/components/ui-libraries/AppLoading";
-import { useFetchUser } from "src/hooks/user/useFetchUser";
 import { FieldInterest, MemberSNSLinks, ProfileImg, University } from "src/components/feature/Member";
 import { ChevronLeftIcon } from "src/components/ui-libraries/icon";
 import { LINKS } from "src/components/utils/constants";
 import { useCurrentUser } from "src/global-states/atoms";
+import { userRepository } from "src/modules/user/user.repository";
 
 export const MemberDetail = () => {
-  const { fetchUser, user, isLoading } = useFetchUser();
+  const { data, isLoading, isError } = useQuery(["findUser"], () => userRepository.findOne(router.query.id as string));
   const { currentUser } = useCurrentUser();
   const router = useRouter();
 
-  useEffect(() => {
-    if (router.isReady) {
-      fetchUser(router.query.id as string);
-    }
-  }, [router.query.id]);
-
   const handleBackClick = () => {
     router.push(LINKS.MEMBER);
   };
 
-  if (isLoading || user == null || currentUser === null) return <AppLoading />;
+  if (isError) return <h1>エラーです</h1>;
+  if (isLoading || data == null || currentUser === null) return <AppLoading />;
```

Comment:
ここの条件だと、fetch失敗した時って、dataがnullだと思うからずっとローディングアイコン表示されちゃいそうやね！

Reply:
ここのfetch失敗したらisErrorがtrueになるから、そもそもこの条件までこないと思う！
なのでイメージ的には

```suggestion
  if (isLoading) return <AppLoading />;
  if (isError) return <h1>エラーです</h1>;
```

だけでいけると思う！

Reply:
dataの型をみたら多分undefined消えてると思う！

Reply:
ほんとだ！助かります！ありがと！

```
  if (isLoading || currentUser === null) return <AppLoading />;
  if (isError) return <h1>エラーです</h1>;
```

これで行きます